frontend-thea-aa/src/lib/components/QualityIndicator.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+    let {quality = 0.0, name = ""} = $props();
+   
+    const ranges = [
+       { max: 0, color: 'bg-black'},    // Green
+       { max: 50, color: 'bg-green-500'},    // Green
+       { max: 100, color: 'bg-yellow-300'},  // Yellow
+       { max: 150, color: 'bg-orange-500'},  // Orange
+       { max: 200, color: 'bg-red-600'},     // Red
+       { max: 300, color: 'bg-purple-600'},  // Purple
+       { max: Infinity, color: 'bg-rose-900'} // Maroon
+    ];
+   
+    let range = ranges.find(r => quality <= r.max) || ranges[ranges.length - 1];
```

Comment:
doesn't matter cause only like 5 ranges but probably can do this with some sort of array indexing solution rather than an object iteration